### PR TITLE
Remove ‘@throws AssertionError’ KDoc from Assertion Functions

### PR DESCRIPTION
**Pull request by jGleitz**
As discussed in #714, those docs where too vague to be actually helpful.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/anyAssertions.kt [hunk 1]
```diff
@@ -11,8 +11,7 @@ import ch.tutteli.kbox.glue
  * Expects that the subject of the assertion is (equal to) [expected].
  *
  * @return An [Expect] for the current subject of the assertion.
- * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
- *
+
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/anyAssertions.kt:14]: I think you forgot to add the `*`, shouldn't it be 
```suggestion
 *
```

Comment (jGleitz) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/anyAssertions.kt:14]: You are right. :disappointed:

Comment (jGleitz) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/anyAssertions.kt]: fixed it.

**Review by robstoll — APPROVED** [review]

**Review by robstoll — CHANGES_REQUESTED** [review]
@jGleitz I change my mind, sorry. I prefer if this commit does not contain changes to deprecated APIs, I am going to remove them in v0.16.0. Can you please revert those changes

**Comment by jGleitz** [comment]
Is it a problem if this _does_ contain changes to deprecated APIs? I did all changes with search + replace. While it’s easy to roll that back for files of deprecated _projects_, finding all deprecated _methods_ would be a lot of manual work.

If those functions will be deleted anyway, why does it matter if we change them in this PR?

**Comment by robstoll** [comment]
It is enough to revert all changes in path misc/deprecated, you don't need to search for methods in api-fluent or api-infix.
I prefer to have a clean history for traceability purposes. The changes to the deprecated API is just noise in this case.

**Comment by jGleitz** [comment]
sure, no problem! Done.

File: core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/AssertionHolder.kt [hunk 2]
```diff
@@ -17,7 +17,6 @@ interface AssertionHolder {
      * @param assertion The assertion which will be added to this holder.
      *
      * @return This holder to support a fluent API.
-     * @throws AssertionError Might throw an [AssertionError] in case the [Assertion] does not hold.
-     */
+         */
```
Comment (robstoll) [inline core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/AssertionHolder.kt:20]: this looks like a formatting problem, but could also be the github diff

File: core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/AssertionPlantWithCommonFields.kt [hunk 3]
```diff
@@ -68,8 +68,7 @@ interface AssertionPlantWithCommonFields<out T> {
          *
          * @param assertions The assertions which shall be checked.
          *
-         * @throws AssertionError Might throw an [AssertionError] if any of the [assertions] does not hold.
-         */
+                 */
```
Comment (robstoll) [inline core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/AssertionPlantWithCommonFields.kt:71]: this looks like a formatting problem, but could also be the github diff

File: core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/Expect.kt [hunk 4]
```diff
@@ -46,8 +46,7 @@ interface Expect<T> : SubjectProvider<T> {
      * @param assertionCreator The receiver function which should create and add assertions to this container.
      *
      * @return An [Expect] for the current subject of the assertion.
-     * @throws AssertionError Might throw an [AssertionError] depending on the concrete implementation.
-     */
+         */
```
Comment (robstoll) [inline core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/Expect.kt:49]: this looks like a formatting problem, but could also be the github diff

File: core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/ReportingAssertionContainer.kt [hunk 5]
```diff
@@ -52,8 +52,7 @@ interface ReportingAssertionContainer<T> : RootExpect<T> {
          *
          * @param assertions The assertions which shall be checked.
          *
-         * @throws AssertionError Might throw an [AssertionError] if any of the [assertions] does not hold.
-         */
+                 */
```
Comment (robstoll) [inline core/api/atrium-core-api-common/src/main/kotlin/ch/tutteli/atrium/creating/ReportingAssertionContainer.kt:55]: this looks like a formatting problem, but could also be the github diff

File: core/robstoll-lib/atrium-core-robstoll-lib-common/src/main/kotlin/ch/tutteli/atrium/core/robstoll/lib/checking/FeatureAssertionChecker.kt [hunk 6]
```diff
@@ -43,8 +43,7 @@ class FeatureAssertionChecker(private val originalAssertionHolder: AssertionHold
      * @param representation Is used as [AssertionGroup.representation].
      * @param assertions Is used as [AssertionGroup.assertions].
      *
-     * @throws AssertionError Might throw an [AssertionError] in case one of the given [assertions] does not hold.
-     */
+         */
```
Comment (robstoll) [inline core/robstoll-lib/atrium-core-robstoll-lib-common/src/main/kotlin/ch/tutteli/atrium/core/robstoll/lib/checking/FeatureAssertionChecker.kt:46]: this looks like a formatting problem, but could also be the github diff

**Review by robstoll — CHANGES_REQUESTED** [review]

**Comment by robstoll** [comment]
Fixed the problems myself. Thanks for the PR :slightly_smiling_face:, looking forward to the others

**Pull request merged**